Mathlib/RepresentationTheory/GroupCohomology/LowDegree.lean

```diff
@@ -331,19 +346,180 @@ theorem oneCoboundaries_eq_bot_of_isTrivial (A : Rep k G) [A.IsTrivial] :
   simp_rw [oneCoboundaries, dZero_eq_zero]
   exact LinearMap.range_eq_bot.2 rfl
 
-theorem mem_twoCoboundaries_of_dOne_apply (x : G → A) :
-    ⟨dOne A x, LinearMap.ext_iff.1 (dTwo_comp_dOne A) x⟩ ∈ twoCoboundaries A :=
-  LinearMap.mem_range_self _ _
+/-- Makes a 2-coboundary out of `f ∈ Im(d¹)`. -/
+def twoCoboundariesOfMemRange (f : G × G → A) (h : f ∈ LinearMap.range (dOne A)) :
+    twoCoboundaries A :=
+  ⟨⟨f, LinearMap.range_le_ker_iff.2 (dTwo_comp_dOne A) h⟩,
+    by rcases h with ⟨x, rfl⟩; exact ⟨x, rfl⟩⟩
 
-theorem mem_twoCoboundaries_of_mem_range (f : G × G → A) (h : f ∈ LinearMap.range (dOne A)) :
-    ⟨f, LinearMap.range_le_ker_iff.2 (dTwo_comp_dOne A) h⟩ ∈ twoCoboundaries A := by
-  rcases h with ⟨x, rfl⟩; exact ⟨x, rfl⟩
+theorem twoCoboundariesOfMemRange_apply (f : G × G → A) {h : f ∈ LinearMap.range (dOne A)} :
+    (twoCoboundariesOfMemRange f h).1.1 = f := rfl
+
+/-- Makes a 2-coboundary out of `f : G × G → A` and `x : G → A` such that
+`ρ(g)(x(h)) - x(gh) + x(g) = f(g, h)` for all `g, h : G`. -/
+def twoCoboundariesOfEq (f : G × G → A) (x : G → A)
+    (hf : ∀ g h, A.ρ g (x h) - x (g * h) + x g = f (g, h)) :
+    twoCoboundaries A :=
+  twoCoboundariesOfMemRange f ⟨x, by ext g; exact hf g.1 g.2⟩
+
+theorem twoCoboundariesOfEq_apply (f : G × G  → A) (x : G → A)
+    {hf : ∀ g h, A.ρ g (x h) - x (g * h) + x g = f (g, h)} :
+    (twoCoboundariesOfEq f x hf).1.1 = f := rfl
 
 theorem mem_range_of_mem_twoCoboundaries (f : twoCocycles A) (h : f ∈ twoCoboundaries A) :
     (twoCocycles A).subtype f ∈ LinearMap.range (dOne A) := by
   rcases h with ⟨x, rfl⟩; exact ⟨x, rfl⟩
 
 end Coboundaries
+section ofDistribMulAction
+section
+variable {G A : Type*} [Mul G] [AddCommGroup A] [SMul G A] (n : ℕ)
+
+/-- A function `f : G → A` satisfies the 1-cocycle condition if
+`f(gh) = g • f(h) + f(g)` for all `g, h : G`. -/
+def IsOneCocycle (f : G → A) : Prop := ∀ g h : G, f (g * h) = g • f h + f g
+
+/-- A function `f : G → A` satisfies the 1-coboundary condition if there's `x : A` such that
+`g • x - x = f(g)` for all `g : G`. -/
+def IsOneCoboundary (f : G → A) : Prop := ∃ x : A, ∀ g : G, g • x - x = f g
+
+/-- A function `f : G × G → A` satisfies the 2-cocycle condition if
+`f(gh, j) + f(g, h) = g • f(h, j) + f(g, hj)` for all `g, h : G`. -/
+def IsTwoCocycle (f : G × G → A) : Prop :=
+  ∀ g h j : G, f (g * h, j) + f (g, h) = g • (f (h, j)) + f (g, h * j)
+
+/-- A function `f : G × G → A` satisfies the 2-coboundary condition if there's `x : G → A` such
+that `g • x(h) - x(gh) + x(g) = f(g, h)` for all `g, h : G`. -/
+def IsTwoCoboundary (f : G × G → A) : Prop :=
+  ∃ x : G → A, ∀ g h : G, g • x h - x (g * h) + x g = f (g, h)
+
+end
+section
+
+variable {k G A : Type u} [CommRing k] [Group G] [AddCommGroup A] [Module k A]
+  [DistribMulAction G A] [SMulCommClass G k A]
+
```

Comment:
Can you restate `mem_oneCocycles_iff` etc. as an iff using the predicates introduced above?

Reply:
Thanks for the comments! I have done this now